Tests/OptimizelyTests-Common/OptimizelyUserContextTests_ODP.swift

```diff
@@ -0,0 +1,281 @@
+// MARK: - Optional parameters
+
+extension OptimizelyUserContextTests_ODP {
+    
+    func testFetchQualifiedSegments_parameters() {
+        try? optimizely.start(datafile: datafile)
+
+        let sem = DispatchSemaphore(value: 0)
+        user.fetchQualifiedSegments(options: [.ignoreCache]) { segments, error in
+            XCTAssertNil(error)
+            XCTAssertEqual(segments, ["odp-segment-1"])
+            sem.signal()
+        }
+        XCTAssertEqual(.success, sem.wait(timeout: .now() + .seconds(3)))
+
+        XCTAssertEqual(kUserId, odpManager.userId, "userId should be used as a default")
+        XCTAssertEqual(Set(["odp-segment-1", "odp-segment-2", "odp-segment-3"]), Set(odpManager.odpConfig.segmentsToCheck), "segmentsToCheck should be all-in-project by default")
+        XCTAssertEqual([.ignoreCache], odpManager.options)
+    }
+    
+    func testFetchQualifiedSegments_configReady() {
+        XCTAssertNil(odpManager.odpConfig.apiKey)
+        XCTAssertNil(odpManager.odpConfig.apiHost)
+
+        try? optimizely.start(datafile: "invalid")
+
+        XCTAssertNil(odpManager.odpConfig.apiKey)
+        XCTAssertNil(odpManager.odpConfig.apiHost)
+        
+        try! optimizely.start(datafile: datafile)
+
+        XCTAssertEqual("W4WzcEs-ABgXorzY7h1LCQ", odpManager.odpConfig.apiKey)
+        XCTAssertEqual("https://api.zaius.com", odpManager.odpConfig.apiHost)
+        
+        let datafile = OTUtils.loadJSONDatafile("decide_datafile")!   // no integration in this datafile
+        try! optimizely.start(datafile: datafile)
+
+        XCTAssertNil(odpManager.odpConfig.apiKey)
+        XCTAssertNil(odpManager.odpConfig.apiHost)
+    }
+    
+}
+
+// MARK: - Tests with live ODP server
+
+// tests below will be skipped in CI (travis/actions) since they use the live ODP server.
+#if DEBUG
+
+extension OptimizelyUserContextTests_ODP {
+    // {"vuid": "00TEST00VUID00FULLSTACK", "fs_user_id": "tester-101"} bound in ODP server for testing
+    var testOdpUserKey: String { return "fs_user_id" }
+    var testOdpUserId: String { return "tester-101"}
+
+    func testLiveOdpGraphQL() {
+        let optimizely = OptimizelyClient(sdkKey: OTUtils.randomSdkKey)
+        try! optimizely.start(datafile: datafile)
+        let user = optimizely.createUserContext(userId: testOdpUserId)
+        
+        let sem = DispatchSemaphore(value: 0)
+        user.fetchQualifiedSegments { segments, error in
+            XCTAssertNil(error)
+            XCTAssertEqual([], segments, "none of the test segments in the live ODP server")
+            sem.signal()
+        }
+        XCTAssertEqual(.success, sem.wait(timeout: .now() + .seconds(30)))
+    }
+    
+    /*
+     this test is not good since createUserContext with not-registered-user will auto register the user.
+     the same live odp test with not-registered-user will be done in ZaiusGraphQLApiManagerTests, so skipped here.
+     
+    func testLiveOdpGraphQL_defaultParameters_userNotRegistered() {
```

Comment:
This will not be included in CI/CD tests, only for local. I need more confidence about ODP server apis for now and future changes. I'll consider to remove it when we have compatible live tests in FSC later.